pkg/state/state_shared_dp.go

```diff
@@ -0,0 +1,129 @@
+/*
+Copyright 2020 NVIDIA
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package state
+
+import (
+	"github.com/pkg/errors"
+	appsv1 "k8s.io/api/apps/v1"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+	"k8s.io/apimachinery/pkg/runtime"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/controller/controllerutil"
+	"sigs.k8s.io/controller-runtime/pkg/source"
+
+	mellanoxv1alpha1 "github.com/Mellanox/network-operator/pkg/apis/mellanox/v1alpha1"
+	"github.com/Mellanox/network-operator/pkg/consts"
+	"github.com/Mellanox/network-operator/pkg/render"
+	"github.com/Mellanox/network-operator/pkg/utils"
+)
+
+// NewStateSharedDp creates a new shared device plugin state
+func NewStateSharedDp(k8sAPIClient client.Client, scheme *runtime.Scheme, manifestDir string) (State, error) {
+	files, err := utils.GetFilesWithSuffix(manifestDir, render.ManifestFileSuffix...)
+	if err != nil {
+		return nil, errors.Wrap(err, "failed to get files from manifest dir")
+	}
+
+	renderer := render.NewRenderer(files)
+	return &stateSharedDp{
+		stateSkel: stateSkel{
+			name:        "state-RDMA-device-plugin",
+			description: "RDMA shared device plugin deployed in the cluster",
+			client:      k8sAPIClient,
+			scheme:      scheme,
+			renderer:    renderer,
+		}}, nil
+}
+
+type stateSharedDp struct {
+	stateSkel
+}
+
+type sharedDpRuntimeSpec struct {
+	Namespace string
+}
+
+type sharedDpManifestRenderData struct {
+	CrSpec      *mellanoxv1alpha1.DevicePluginSpec
+	RuntimeSpec *sharedDpRuntimeSpec
+}
+
+// Sync attempt to get the system to match the desired state which State represent.
+// a sync operation must be relatively short and must not block the execution thread.
+func (s *stateSharedDp) Sync(customResource interface{}, infoCatalog InfoCatalog) (SyncState, error) {
+	cr := customResource.(*mellanoxv1alpha1.NicClusterPolicy)
+	log.V(consts.LogLevelInfo).Info(
+		"State: state-RDMA-device-plugin Sync Custom resource", "Name", cr.Name, "Namespace:", cr.Namespace)
```

Comment:
use constant for tate-RDMA-device-plugin same for Line 44

Reply:
will be done in next update to the PR